src/param_env.md

```diff
@@ -0,0 +1,17 @@
+# Parameter Environment
+
+When working with associated constants or generic types it is often relevant to
+have more information about the `Self` or generic parameters. Trait bounds and
```

Comment:
I think the easiest way to explain the parameter environment is that it represents the set of where clauses in scope. I would also always try to tie to an example:

```rust
fn foo<T: Copy>(t: T) {
}
```

when you request `tcx.param_env(def_id_for_foo)` you get back `[T: Copy]`, basically.